quickstart-node/src/routes/auth.route.ts

```diff
@@ -0,0 +1,69 @@
+import express from 'express';
+import { Request, Response } from 'express';
+import { BadRequest, InternalServerError } from 'http-errors';
+import { EnvironmentVariablesService, SecureStorageService } from '@services/monday-code';
+import { JWTService } from '@services/jwt.service';
+import { mondayRequestAuth } from '@middlewares/auth.middleware';
+import { generateUrlSafeString } from '@utils/url.utils';
+import { EnvironmentsKeys } from '@shared/monday-auth-consts';
+
+const authRouter = express.Router();
+
+// Validates the state parameter from the OAuth2 callback.
+const validateState = (req: Request) => {
+  const returnedState = req.query.state as string;
+  const savedState = req.cookies?.state;
+  if (returnedState !== savedState) {
+    throw new BadRequest('Invalid state parameter');
+  } else if (!req.cookies?.user_id) {
+    throw new InternalServerError('"user_id" is not defined');
+  }
+};
+
+authRouter.get('/', mondayRequestAuth, async (req: Request, res: Response) => {
+  const userId = String(req.session?.userId);
+  const backToUrl = req.session?.backToUrl;
+
+  const connection = await SecureStorageService.getInstance().get<any>(userId);
+  if (connection && connection.monday_token) {
+    return res.redirect(backToUrl);
+  }
+
+  await SecureStorageService.getInstance().set(userId, { back_to_url: backToUrl });
+
+  // Generate a random state for CSRF protection
+  const state = generateUrlSafeString(16);
+
+  // Prepare OAuth2 parameters
+  const params = new URLSearchParams({
+    client_id: EnvironmentVariablesService.getInstance().get(EnvironmentsKeys.MONDAY_OAUTH_CLIENT_ID) as string,
+    state: state,
+  });
+
+  const redirectUrl = new URL(
+    EnvironmentVariablesService.getInstance().get(EnvironmentsKeys.MONDAY_OAUTH_BASE_PATH) as string,
```

Comment:
Maybe add some error if there's no MONDAY_OAUTH_BASE_PATH, other then to re-route to undefined, but not super critical

Reply:
It is a critical error, so what will happen, it will get to the error middleware and send `res.status(500)` with internal server error message. And the specific error will be logged to the server.
I don't think re-route to undefined is the case here, sending an internal server error is the right outcome for this error.
What do you think?